### PR TITLE
some fixes for flow_alugrid

### PR DESCRIPTION
**Pull request by ElyesAhmed**
Added instantiation to initstateequil.cc and correct invocation of AquiferModel, both needed for flow_alugrid

### CODE REVIEW

File: ebos/equil/initstateequil.cc [hunk 1]
```diff
@@ -1901,12 +1906,29 @@ using GridView = Dune::Fem::GridPart2GridViewImpl<
...
 #endif
 
+#if HAVE_DUNE_ALUGRID
+
+#if HAVE_MPI
+    using ALUGrid3CN = Dune::ALUGrid<3, 3, Dune::cube, Dune::nonconforming, Dune::ALUGridMPIComm>;
+#else
+    using ALUGrid3CN = Dune::ALUGrid<3, 3, Dune::cube, Dune::nonconforming, Dune::ALUGridNoComm>;
+#endif //HAVE_MPI
+#endif //HAVE_DUNE_ALUGRID
```
Comment (bska) [inline ebos/equil/initstateequil.cc:1916]: Any reason not to put this together with the other `HAVE_DUNE_ALUGRID` section&mdash;i.e., the one that instantiates the `InitialStateComputer<>` template for ALUGrid?  On a separate note, I might actually write this bit as
```C++
#if HAVE_MPI
using ALUGridComm = Dune::ALUGridMPIComm;
#else
using ALUGridComm = Dune::ALUGridNoComm;
#endif // HAVE_MPI

using ALUGrid3CN = Dune::ALUGrid<3, 3, Dune::cube, Dune::nonconforming, ALUGridComm>;
```
to make it visually clear that the only difference between the two alternatives is the communicator.

File: flow/flow_blackoil_alugrid.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace Opm::Properties {
 // by default use the dummy aquifer "model"
 template<class TypeTag>
 struct EclAquiferModel<TypeTag, TTag::EclFlowProblemAlugrid> {
-    using type = Opm::EclBaseAquiferModel<TypeTag>;
+    using type = Opm::BlackoilAquiferModel<TypeTag>;
```
Comment (bska) [inline flow/flow_blackoil_alugrid.cpp:33]: Why does the `BaseAquiferModel` not meet ALUGrid's requirements?

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
Thanks for the fixes.  I only have a couple of questions about the details here and then I'm prepared to merge this if the build check is green.

File: ebos/equil/initstateequil.cc [hunk 3]
```diff
@@ -1942,9 +1925,22 @@ template InitialStateComputer<BlackOilFluidSystem<double>,
...
-                         const bool);
-
+                         const bool);                                    
 #if HAVE_DUNE_ALUGRID
+#if HAVE_MPI
+using ALUGridComm = Dune::ALUGridMPIComm;
+#else
+using ALUGridComm = Dune::ALUGridNoComm;
+#endif //HAVE_MPI
+using ALUGrid3CN = Dune::ALUGrid<3, 3, Dune::cube, Dune::nonconforming, Dune::ALUGridMPIComm>;
```
Comment (akva2) [inline ebos/equil/initstateequil.cc:1935]: wrong.

**Review by bska — COMMENTED** [review]
Thanks for the updates.  This looks good to me for the most part, with the possible exception of the new `addToSource()` function in `EclBaseAquiferModel`.  Were you not able to use the signature from `BlackoilAquiferModel` here?

**Comment by bska** [comment]
jenkins build this please

**Comment by ElyesAhmed** [comment]
> Thanks for the updates. This looks good to me for the most part, with the possible exception of the new `addToSource()` function in `EclBaseAquiferModel`. Were you not able to use the signature from `BlackoilAquiferModel` here?

I fixed that by avoiding using the dummy aquifer model! I hope that's Ok now, Thanks

**Comment by bska** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Comment by ElyesAhmed** [comment]
thanks!

**Pull request merged**